3.workload.md

```diff
@@ -0,0 +1,99 @@
+# 3. Workload
+
+This section defines the workload definition schematics.
+
+Usually, workload definitions are provided by the OAM runtime (just as traits are) so that application developers may inspect the OAM runtime and learn what workloads are available for use.  
+
+## Top-Level Attributes
+
+The following attributes are common across all schemata defined in this document. They follow the [Kubernetes specification](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields).
+
+Here are the attributes that provide top-level information about the workload definition.
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `apiVersion` | `string` | Y | | The specific version of the specification in use. The core types use `core.oam.dev/v1alpha2` |
+| `kind` | `string` | Y || The string `WorkloadDefinition` |
+| `metadata` | [`Metadata`](#metadata) | Y | | WorkloadDefinition metadata. |
+| `spec`| [`Spec`](#spec) | Y | | A container for the workload definition. |
+
+### Metadata
+
+The metadata describes this workload definition. The `metadata` section is defined in [the meta section of part 2](2.overview_and_terminology.md#Metadata)
+
+### Spec
+
+The specification section contains a reference to the workload definition.
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `definitionRef` | [`DefinitionRef`](#DefinitionRef) | Y | | Workload schema reference. |
+
+
+#### DefinitionRef
+
+DefinitionRef defines an index to find the workload schema.
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `schemaName` | `string` | Y | | The workload schema's name. |
+
+The workload schema definition itself __MUST__ contain information that can be used to uniquely identify it. OAM runtime will use that information in a [Component](4.component.md) to refer to the workload schema instead of using the name of the workload definition.
+ We recommend to use the format [Group/Version/Kind](2.overview_and_terminology.md#Group) described in section 2 as the unique identification in the schema. 
```

Comment:
> ```yaml
> spec:
>   definitionRef:
>     name: core.oam.dev/v1alpha1.ContainerizedWorkload
> ```
> 
> Is this recommending that runtimes use the above `group/version.Kind` syntax to uniquely identify a schema? I think this would be surprising and misleading to folks familiar with Kubernetes, and make implementing the spec a little more difficult for runtimes that are built using Kubernetes.
> 
> Quoting API conventions:
> 
> > The name of a field referring to another resource of kind Foo by name should be called fooName. The name of a field referring to another resource of kind Foo by ObjectReference (or subset thereof) should be called fooRef.
> 
> > Object references should either be called fooName if referring to an object of kind Foo by just the name (within the current namespace, if a namespaced resource), or should be called fooRef, and should contain a subset of the fields of the ObjectReference type.
> 
> i.e. `spec.fooRef.name` should refer to the `metadata.name` of a `kind: Foo`. In the case of `definitionRef` we omit the kind and leave it up to the runtime (perhaps it's a `kind: CustomResourceDefinition`, perhaps it's a `kind: SweetXMLSchema`) but for Kubernetes runtimes I think this should be the _name_ of the definition, not a GVK. The `group/version.Kind` syntax is not a valid Kubernetes name (due to the `/` and uppercase kind), and CRDs specifically [MUST](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition) be named plural.group, e.g. `containerizedworkloads.oam.core.dev`.
> 
> So TL;DR I expect all runtimes that are implemented atop the Kubernetes API server would need to write:
> 
> ```yaml
> spec:
>   definitionRef:
>     name: containerisedworkloads.oam.core.dev
> ```
No, that link actually points to exactly what you describe here